src/prefect/runtime/task_run.py

```diff
@@ -83,6 +83,14 @@ def get_tags() -> List[str]:
         return task_run_ctx.task_run.tags
 
 
+def get_run_count() -> int:
+    task_run_ctx = TaskRunContext.get()
+    if task_run_ctx is None:
```

Comment:
What's the reason behind not getting the run count from the api here similar to flow run if there is no available ctx?

Reply:
@jakekaplan If we are not in a task run context and we are using `runtime.task_run`, I think it may difficult to discern the task run id in order to retrieve attributes from it. 

On the flip side, when we use `runtime.flow_run`, we only create the call when the flow run context is `None` but we have a `run_id` (from being in a task run context or from an env var). 

Said another way, if we are in a task run context when we are calling `runtime.flow_run`, we can retrieve the flow run id from a variety of places, such as the flow run context, task run context, or env var. However, when we call `runtime.task_run` in a task run context, we attempt to retrieve it that way, but if we aren't in one, I don't know of how else to identify what task we're talking about.

Please let me know if that doesn't make sense!

Reply:
thanks for the explanation! That makes a lot of sense to me